packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -357,9 +360,39 @@ export class SearchPage extends Component {
       return;
     }
 
+    const data = localStorage.getItem(standaloneSearchBoxStorageKey);
+
+    if (data) {
+      this.executeFirstSearchAfterStandaloneSearchBoxRedirect(data);
+      return;
+    }
+
     this.engine.executeFirstSearch();
   }
 
+  private executeFirstSearchAfterStandaloneSearchBoxRedirect(data: string) {
+    localStorage.removeItem(standaloneSearchBoxStorageKey);
+    const parsed: {
+      value: string;
+      analytics: StandaloneSearchBoxAnalytics;
+    } = JSON.parse(data);
+    const {value, analytics} = parsed;
+    const {cause, metadata} = analytics;
+
+    const {updateQuery} = loadQueryActions(this.engine);
+    const {logOmniboxFromLink, logSearchFromLink} = loadSearchAnalyticsActions(
+      this.engine
+    );
+
+    const event =
+      metadata && cause === 'omniboxFromLink'
+        ? logOmniboxFromLink(metadata)
+        : logSearchFromLink();
+
+    this.engine.dispatch(updateQuery({q: value}));
+    this.engine.executeFirstSearch(event);
```

Comment:
I'm debating adding a new method on the search-engine to handle the selected lines, since it's still quite a bit of logic to get things right. If a method was exposed, it would become:

```
const json = localStorage.getItem(standaloneSearchBoxStorageKey);

if (json) {
      localStorage.removeItem(standaloneSearchBoxStorageKey);
      const data = JSON.parse(json);
      engine.executeFirstSearchAfterStandaloneSearchBoxRedirect(data);
      return;
    }
```

Reply:
I'd be for a method,but the name could be shorter 😄